mobile/library/cc/stream_prototype.cc

```diff
@@ -10,10 +10,11 @@ StreamPrototype::StreamPrototype(EngineSharedPtr engine) : engine_(engine) {
 }
 
 StreamSharedPtr StreamPrototype::start(bool explicit_flow_control) {
-  auto envoy_stream = init_stream(engine_->engine_);
-  start_stream(engine_->engine_, envoy_stream, callbacks_->asEnvoyHttpCallbacks(),
+  envoy_engine_t engine_handle = reinterpret_cast<envoy_engine_t>(engine_->engine_);
```

Comment:
since this gets used by the MainInterface which calls the Http::Client (which is all c++), can we do away with the engine_handle and just send `Engine*`?

Reply:
soon - I was getting rid of the "engine" functions in this call.  I started the follow-up PR to remove the stream functions but those are more complicated because they're not simple wrappers - they also do the "post to dispatcher" thread context switch.  To remove those we have to change the client to do that context switch which may break unit tests so I decided to break into a separate PR.

Reply:
makes sense, thanks for the explanation!